src/app/application/components/app-collection/AppCollection.vue

```diff
@@ -219,7 +219,12 @@ function getRowAttributes(row: Row): Record<string, string> {
 const click = (e: MouseEvent) => {
   const $tr = (e.target as HTMLElement).closest('tr')
   if ($tr) {
-    const $a: HTMLAnchorElement | null = $tr.querySelector('td:first-child a')
+    const $a: HTMLAnchorElement | null = ['td:first-child a', '[data-action]'].reduce<HTMLAnchorElement | null>((prev, item) => {
+      if (prev === null) {
+        return $tr.querySelector(item)
+      }
+      return prev
+    }, null)
```

Comment:
**Question**: How come this code is becoming more complex by the week? It should become simpler by selecting a specifically marked element and then being done with it.

See https://github.com/kumahq/kuma-gui/issues/2275 and https://github.com/kumahq/kuma-gui/issues/2293.

Reply:
AppCollection is going to be completely redone soon, and I would think that the AppCollection component will be replaced by an `XTable` that does less.

Just that won't happen before 2.7 goes out.

For the moment we just support two ways i.e. if it can't find an anchor in the first cell, then look for a `[data-action]` instead. That sort of 'complexity' I'm more than ok with, especially seeing as this code will soon be removed from here entirely seeing as we need this functionality in other places apart from AppCollection.